### ISSUE TITLE
[no-shadow] False positive for enum

### ISSUE THREAD

**Issue by restrry** [typescript-eslint/typescript-eslint#2360]
**Repro**

I created a repo to reproduce the problem https://github.com/restrry/typescript-eslint-scope-bug

```JSON
{
  "rules": {
    "@typescript-eslint/no-shadow": "error"
  }
}
```

```TS
enum Direction {
    left = 'left',
    right = 'right',
}
```

**Expected Result**
No errors reported.

**Actual Result**
`  1:6  error  'Direction' is already declared in the upper scope  @typescript-eslint/no-shadow`

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.0.0-alpha.5` |
| `@typescript-eslint/parser`        | `4.0.0-alpha.5` |
| `TypeScript`                       | `3.9.5` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `10.21.0` |
| `npm`                              | `N/A` |
| `yarn`                              | `1.21.1` |
